scripts/pin-rust-nightly.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+if [ "$#" -ge 1 ]; then
+    echo "Trying to switch to date `$1`, please check, that format is `yyyy-mm-dd`"
+else
+    echo "No date provided"
+    exit 1
+fi
+
+pin_date=$1
+os_name="$(uname)"
+
+if [ "$os_name" = "Darwin" ]; then
+    rustup toolchain install nightly-$pin_date --component llvm-tools-preview
+    rustup target add wasm32-unknown-unknown --toolchain nightly-$pin_date
+    rm -rf ~/.rustup/toolchains/nightly-aarch64-apple-darwin
```

Comment:
This will not work on x84 macOS. I'd better use target-triple info from the compiler itself. E.g.:

```shell
❯ rustc -Vv | grep "host: " | sed "s/^host: \(.*\)$/\1/"
aarch64-apple-darwin
```

Reply:
ok add identifying